src/web-auth/cross-origin-authentication.js

```diff
@@ -87,29 +99,47 @@ CrossOriginAuthentication.prototype.login = function(options, cb) {
           error: 'request_error',
           error_description: JSON.stringify(err)
         };
+
         return responseHandler(cb, { forceLegacyError: true })(errorObject);
       }
-      var popupMode = options.popup === true;
-      options = objectHelper.blacklist(options, [
-        'password',
-        'credentialType',
-        'otp',
-        'popup'
-      ]);
-      var authorizeOptions = objectHelper
-        .merge(options)
-        .with({ loginTicket: data.body.login_ticket });
-      var key = createKey(_this.baseOptions.rootUrl, data.body.co_id);
-      _this.storage.setItem(key, data.body.co_verifier, {
-        expires: times.MINUTES_15
-      });
-      if (popupMode) {
-        _this.webMessageHandler.run(
-          authorizeOptions,
-          responseHandler(cb, { forceLegacyError: true })
-        );
+
+      function doAuth() {
+        var popupMode = options.popup === true;
+
+        options = objectHelper.blacklist(options, [
+          'password',
+          'credentialType',
+          'otp',
+          'popup',
+          'onRedirecting'
+        ]);
+
+        var authorizeOptions = objectHelper
+          .merge(options)
+          .with({ loginTicket: data.body.login_ticket });
+
+        var key = createKey(_this.baseOptions.rootUrl, data.body.co_id);
+
+        _this.storage.setItem(key, data.body.co_verifier, {
+          expires: times.MINUTES_15
+        });
+
+        if (popupMode) {
```

Comment:
When `popupMode` is true, is there an actual redirect involved?

Reply:
Well, the "redirect" happens by way of the web message flow and a hidden iframe, but this code executes after the `onRedirecting` hook has already happened anyway. Make sense?